cmake/CMakeLists.txt

```diff
@@ -322,14 +322,22 @@ if (WIN32)
         string(APPEND CMAKE_CXX_FLAGS " /W4")
     endif()
 
-    #only treat warning as error on x64 platform. For x86, right no there are too many warnings to fix
+    # treat warning as error only on x64 platform. For x86, there are too many warnings to fix.
     if (CMAKE_SIZEOF_VOID_P EQUAL 8 AND onnxruntime_DEV_MODE)
       # treat warnings as errors
       string(APPEND CMAKE_CXX_FLAGS " /WX")
       foreach(type EXE STATIC SHARED)
         set(CMAKE_${type}_LINKER_FLAGS "${CMAKE_${type}_LINKER_FLAGS} /WX")
       endforeach()
     endif()
+
+    # set linker flags to minimize the binary size.
+    if (MSVC)
+      foreach(type EXE SHARED)    
+        set(CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO} /OPT:REF")
```

Comment:
How about also set it for the release build?

set(CMAKE_${type}_LINKER_FLAGS_RELEASE ....)

Reply:
It makes no difference for release. So I played conservative.